functional-test/build.gradle

```diff
@@ -62,13 +62,47 @@ def checkWarFile(destination, fileName) {
     println "File copied from ${warFile} to ${destination + fileName}"
 }
 
+/** 
+ * Returns a function that includes only the test files specified in the `testFiles` parameter.
+ * The `testFiles` parameter is a string of paths to the source code of the tests to be run.
+ */
+def explicitTestIncluder(String testFiles) {
+    // Convert test file source code paths to compiled class paths.
+    // Example: `functional-test/src/test/groovy/org/rundeck/tests/functional/api/basic/BasicSpec.groovy`
+    // needs to become `org/rundeck/tests/functional/api/basic/BasicSpec.class`.
```

Comment:
I presume it was tried but i must ask: I wonder if there's a simpler way to get this information from gradle.

Reply:
As far as I can tell this is not possible while hooking into the [include](https://docs.gradle.org/7.6.2/javadoc/org/gradle/api/tasks/testing/Test.html#include-groovy.lang.Closure-) function of Gradle test tasks. The [FileTreeElements](https://docs.gradle.org/7.6.2/javadoc/org/gradle/api/file/FileTreeElement.html) point to the compiled test classes rather than source code.

Reply:
I guess an alternative approach might have been to modify Gradle's source sets but that might have other challenges?